server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -528,4 +528,8 @@ public void advanceMaxSeqNoOfUpdatesOrDeletes(long maxSeqNoOfUpdatesOnPrimary) {
         assert maxSeqNoOfUpdatesOnPrimary <= getMaxSeqNoOfUpdatesOrDeletes() :
             maxSeqNoOfUpdatesOnPrimary + ">" + getMaxSeqNoOfUpdatesOrDeletes();
     }
+
+    protected static DirectoryReader openDirectory(Directory dir) throws IOException {
+        return new SoftDeletesDirectoryReaderWrapper(DirectoryReader.open(dir, OFF_HEAP_READER_ATTRIBUTES), Lucene.SOFT_DELETES_FIELD);
```

Comment:
In 7.x and below, this will be dependent on the soft-deletes setting, right?

Reply:
Yes